examples/minimal-winit/src/main.rs

```diff
@@ -4,137 +4,147 @@
-    error!("{method_name}() failed: {err}");
-    for source in err.sources().skip(1) {
-        error!("  Caused by: {source}");
-    }
-}
-
 impl World {
-    /// Create a new `World` instance that can draw a moving box.
     fn new() -> Self {
-        Self {
-            box_x: 24,
-            box_y: 16,
-            velocity_x: 1,
-            velocity_y: 1,
-        }
+        Self { box_x: 24, box_y: 16, velocity_x: 1, velocity_y: 1 }
     }
-
-    /// Update the `World` internal state; bounce the box around the screen.
     fn update(&mut self) {
         if self.box_x <= 0 || self.box_x + BOX_SIZE > WIDTH as i16 {
             self.velocity_x *= -1;
         }
         if self.box_y <= 0 || self.box_y + BOX_SIZE > HEIGHT as i16 {
             self.velocity_y *= -1;
         }
-
         self.box_x += self.velocity_x;
         self.box_y += self.velocity_y;
     }
-
-    /// Draw the `World` state to the frame buffer.
-    ///
-    /// Assumes the default texture format: `wgpu::TextureFormat::Rgba8UnormSrgb`
     fn draw(&self, frame: &mut [u8]) {
-        for (i, pixel) in frame.chunks_exact_mut(4).enumerate() {
+        for (i, px) in frame.chunks_exact_mut(4).enumerate() {
             let x = (i % WIDTH as usize) as i16;
             let y = (i / WIDTH as usize) as i16;
-
-            let inside_the_box = x >= self.box_x
+            let inside = x >= self.box_x
                 && x < self.box_x + BOX_SIZE
                 && y >= self.box_y
                 && y < self.box_y + BOX_SIZE;
+            let rgba = if inside { [0x5e, 0x48, 0xe8, 0xff] } else { [0x48, 0xb2, 0xe8, 0xff] };
+            px.copy_from_slice(&rgba);
+        }
+    }
+}
+
+fn log_error<E: std::error::Error + 'static>(method_name: &str, err: E) {
+    error!("{method_name}() failed: {err}");
+    for source in err.sources().skip(1) {
+        error!("  Caused by: {source}");
+    }
+}
+
+struct App {
+    window: Option<&'static Window>,     // leaked &'static Window
+    pixels: Option<Pixels<'static>>,     // borrows the window
+    world: World,
+}
+
+impl App {
+    fn new() -> Self {
+        Self { window: None, pixels: None, world: World::new() }
+    }
+}
 
-            let rgba = if inside_the_box {
-                [0x5e, 0x48, 0xe8, 0xff]
-            } else {
-                [0x48, 0xb2, 0xe8, 0xff]
-            };
+impl ApplicationHandler for App {
+    fn resumed(&mut self, elwt: &ActiveEventLoop) {
+        if self.window.is_none() {
+            // Create window
+            let size = LogicalSize::new(WIDTH as f64, HEIGHT as f64);
+            let attrs = WindowAttributes::default()
+                .with_title("Hello Pixels")
+                .with_inner_size(size)
+                .with_min_inner_size(size);
+            let window = elwt.create_window(attrs).expect("create window");
+
+            // Leak to get &'static Window so Pixels can live in the struct
+            let window: &'static Window = Box::leak(Box::new(window));
```

Comment:
There is no need to leak anything to satisfy the `'static` constraint. Just use `Arc`. See: https://github.com/parasyte/pixels/issues/403#issuecomment-2405750064 and discussion in https://github.com/parasyte/pixels/issues/411